@xen-orchestra/lite/src/components/Table.tsx

```diff
@@ -19,55 +54,320 @@ interface ParentState {}
+            required
+            sx={{
+              backgroundColor: '#E8E8E8',
+              border: 'none',
+              color: '#007bff',
+              height: '24px',
+            }}
+            value={showByValue}
+            valueRenderer={item => item}
+          />
+        </StyledPaginationFlexRow>
+      </StyledPaginationContainer>
+    </Box>
+  )
+)
+
+const Table = withState<State, Props, Effects, Computed, ParentState, ParentEffects>(
+  {
+    effects: {
+      handlePaginationChange: function (_, page) {
+        const reg = new RegExp(this.props.stateUrlParam + '_page=\\d+')
+        let searchParams = this.props.location.search
+        if (reg.test(searchParams)) {
+          searchParams = searchParams.replace(reg, `${this.props.stateUrlParam}_page=${page}`)
+        } else {
+          searchParams += `?${this.props.stateUrlParam}_page=${page}`
+        }
+        this.props.history.push(`${this.props.location.pathname}${searchParams}`)
+      },
+      handleRowPerPage: function (e) {
+        const reg = new RegExp(this.props.stateUrlParam + '_row=\\d+')
+        let searchParams = this.props.location.search
+        if (reg.test(searchParams)) {
+          searchParams = searchParams.replace(reg, `${this.props.stateUrlParam}_row=${e.target.value}`)
+        } else {
+          searchParams += `?${this.props.stateUrlParam}_row=${e.target.value}`
+        }
+        this.props.history.push(`${this.props.location.pathname}${searchParams}`)
+      },
+      toggleAllSelectedItem: function () {
+        if (this.props.collection === undefined) {
+          return
+        }
+        this.props.onSelectItems?.(
+          this.props.selectedItems.length !== this.props.collection.length ? this.props.collection : []
+        )
+      },
+      toggleSelectItem: function (item: any) {
+        const { selectedItems } = this.props
+        if (selectedItems === undefined) {
+          return
+        }
+        const index = findIndex(selectedItems, item)
+        index === -1 ? selectedItems.push(item) : selectedItems.splice(index, 1)
+        this.props.onSelectItems(selectedItems)
+      },
+    },
+    computed: {
+      page: (_, { location, stateUrlParam }) =>
+        Number(location.search.match(new RegExp(stateUrlParam + '_page=\\d+'))?.[0]?.split('=')[1] ?? 0),
+      paginatedCollection: ({ page, rowsPerPage }, { collection }) =>
+        collection?.slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage),
+      rowsPerPage: (_, { location, stateUrlParam, rowsPerPageOptions }) => {
+        const _rowsPerPageOptions = rowsPerPageOptions ?? [10, 25, 50, 100]
+        let rowsPerPage = Number(
+          location.search.match(new RegExp(stateUrlParam + '_row=\\d+'))?.[0]?.split('=')[1] ?? _rowsPerPageOptions[0]
+        )
+        if (!_rowsPerPageOptions.includes(rowsPerPage)) {
+          console.error('Not allowed rowsPerPage', rowsPerPage)
+          rowsPerPage = _rowsPerPageOptions[0]
+        }
```

Comment:
Is this really necessary? Is it an issue to let the user craft a URL with a custom "items per page" value?